query/src/main/java/org/infinispan/query/affinity/AffinityErrorHandler.java

```diff
@@ -0,0 +1,56 @@
+package org.infinispan.query.affinity;
+
+import org.apache.lucene.store.LockObtainFailedException;
+import org.hibernate.search.backend.LuceneWork;
+import org.hibernate.search.exception.ErrorContext;
+import org.hibernate.search.exception.ErrorHandler;
+import org.hibernate.search.exception.impl.LogErrorHandler;
+import org.infinispan.query.backend.WrappingErrorHandler;
+import org.infinispan.query.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @since 9.0
+ */
+public class AffinityErrorHandler extends WrappingErrorHandler {
+
+   private static final Log log = LogFactory.getLog(AffinityErrorHandler.class, Log.class);
+
+   private OperationFailedHandler operationFailedHandler;
+
+   public AffinityErrorHandler(ErrorHandler handler) {
+      super(handler);
+   }
+
+   public AffinityErrorHandler() {
+      super(new LogErrorHandler());
+   }
+
+   public void initialize(OperationFailedHandler operationFailedHandler) {
+      if (log.isDebugEnabled()) log.debugf("Initializing with %s", operationFailedHandler);
+      this.operationFailedHandler = operationFailedHandler;
+   }
+
+   @Override
+   protected void errorOccurred(ErrorContext context) {
```

Comment:
this will be logged via the wrapped Hibernate Search's ```ErrorHandler```, so no point in logging it again